pkg/segment/structs/evaluationstructs.go

```diff
@@ -1038,32 +1068,39 @@ func (self *ValueExpr) EvaluateToString(fieldToValue map[string]utils.CValueEncl
 		if err != nil {
 			//Because parsing is successful and it can not evaluate as a float in here,
 			//There is one possibility: field name may not be float
+			if !self.NumericExpr.IsTerminal {
+				// But this is not a terminal case, so we just return 0
+				return "0", err
+			}
```

Comment:
I don't understand why we should return `"0"` instead of `""` in this error case. Also, since we return an error, does it matter what the string is?